packages/NodeTypeResolver/PHPStan/Scope/PHPStanNodeScopeResolver.php

```diff
@@ -74,6 +75,11 @@ public function processNodes(array $stmts, SmartFileInfo $smartFileInfo): array
 
         // skip chain method calls, performance issue: https://github.com/phpstan/phpstan/issues/254
         $nodeCallback = function (Node $node, MutatingScope $mutatingScope) use (&$nodeCallback): void {
+            if ($node instanceof Foreach_) {
+                // decorate value as well
+                $node->valueVar->setAttribute(AttributeKey::SCOPE, $mutatingScope);
+            }
+
```

Comment:
@ondrejmirtes Hi Ondra, btw, we had some weird bug with missing scope. I though it's Rector creating new nodes that do not have it, but in the end it showed the `Foreach->var` has no `Scope`. The hotfix is to add it manually here.

Is that on purpose to walk around something or just missing?